web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx

```diff
@@ -518,19 +613,216 @@ class HtxParagraphsView extends Component {
+      this.hotkey.addNamed("phrases:select_all_annotate", this.handleSelectAllAndAnnotate);
+
+      this.hotkey.addNamed("phrases:next-region", this.handleNextRegion);
+      this.hotkey.addNamed("phrases:previous-region", this.handlePreviousRegion);
+    }
   }
 
   componentWillUnmount() {
     const target = document.querySelector(this.mainContentSelector);
 
     if (target) this._resizeObserver?.unobserve(target);
     this._resizeObserver?.disconnect();
+    if (isFF(FF_NER_SELECT_ALL)) this.props.item.setViewRef(null);
+    if (this.hotkey) this.hotkey.unbindAll();
+  }
+
+  // Check if any labels are selected for the current annotation (reactive to MobX changes)
+  get hasSelectedLabels() {
+    if (!isFF(FF_NER_SELECT_ALL)) return false;
+
+    try {
+      const { item } = this.props;
+      const states = item.activeStates && item.activeStates();
+
+      console.log("[DEBUG] hasSelectedLabels - activeStates:", {
+        itemName: item.name,
+        hasActiveStatesMethod: typeof item.activeStates,
+        states: states,
+        statesLength: states?.length || 0,
+        hasStates: !!(states && states.length > 0),
+      });
+
+      return !!(states && states.length > 0);
+    } catch (error) {
+      console.warn("Error checking selected labels:", error);
+      return false;
+    }
   }
 
+  selectText = (phraseIndex) => {
+    const item = this.props.item;
+    const phrases = item._value;
+    const cls = item.layoutClasses;
+    const phraseElements = this.myRef.current?.getElementsByClassName(cls.text);
+
+    if (!phrases || phraseIndex < 0 || phraseIndex >= phrases.length || !phraseElements) return;
+
+    const phraseElement = phraseElements[phraseIndex];
+    if (!phraseElement) return;
+
+    const range = document.createRange();
+    range.selectNodeContents(phraseElement);
+    const selection = window.getSelection();
+    selection.removeAllRanges();
+    selection.addRange(range);
+  };
+
   setIsInViewPort(isInViewPort) {
     this.setState({ inViewPort: isInViewPort });
   }
 
+  // Helper to deselect only regions without affecting labels
+  unselectAllRegions = () => {
+    const item = this.props.item;
+    if (!item || !item.annotation) return;
+
+    // Get currently selected regions
+    const selectedRegions = item.annotation.selectedRegions || [];
+
+    // Deselect each region individually to preserve label selection
+    selectedRegions.forEach((region) => {
+      if (item.annotation.unselectArea && typeof item.annotation.unselectArea === "function") {
+        item.annotation.unselectArea(region);
+      }
+    });
+  };
+
+  // Helper to select all regions for a phrase index
+  selectRegionsForPhrase = (phraseIdx) => {
+    if (!isFF(FF_NER_SELECT_ALL)) return;
+
+    const item = this.props.item;
+    if (!item || !item.annotation || !item.annotation.results) return;
+
+    // Only deselect regions, not labels
+    this.unselectAllRegions();
```

Comment:
we already have this method
```suggestion
    item.annotation.unselectAreas();
```